### PR TITLE
ChangeUser with SDK Authentication implementation.

### PR DESCRIPTION
**Pull request by spaluchiewicz**
Expose available in native implementation (both for **iOS** and **Android**) API to use SDK Auth feature.

### CODE REVIEW

File: android/src/main/kotlin/com/braze/brazeplugin/BrazePlugin.kt [hunk 1]
```diff
@@ -162,7 +162,18 @@ class BrazePlugin : MethodCallHandler, FlutterPlugin, ActivityAware {
             when (call.method) {
                 "changeUser" -> {
                     val userId = call.argument<String>("userId")
-                    Braze.getInstance(context).changeUser(userId)
+                    val sdkAuthSignature = call.argument<String>("sdkAuthSignature")
+                    if (sdkAuthSignature != null) {
+                        Braze.getInstance(context).changeUser(userId, sdkAuthSignature)
+                    } else {
+                        Braze.getInstance(context).changeUser(userId)
```
Comment (bryanlogan) [inline android/src/main/kotlin/com/braze/brazeplugin/BrazePlugin.kt:169]: changeUser(userId) will just turn around and call changeUser(userId, null), so we could simplify this a little bit.

Comment (spaluchiewicz) [reply android/src/main/kotlin/com/braze/brazeplugin/BrazePlugin.kt]: @bryanlogan updated according to the comment.

**Review by bryanlogan — APPROVED** [review]

**Comment by hokstuff** [comment]
Hi @spaluchiewicz,

Thanks for submitting this PR! We will integrate your changes into our private repo to include in our release process, and then we'll give you a shoutout once it is completed and released. Stay tuned!

**Comment by hokstuff** [comment]
Hi @spaluchiewicz,

Thank you for your contribution! We have consolidated the changes in this PR into our latest [Flutter SDK 2.5.0 release](https://github.com/braze-inc/braze-flutter-sdk/releases/tag/2.5.0) with a few slight changes. Feel free to point to that version and let us know if you run into any issues!

**Pull request closed without merging**